src/twiliosim.cr

```diff
@@ -31,28 +30,81 @@ class Twiliosim::Server
       response = {"sid" => UUID.random().to_s()}
       response.to_json(context.response)
     when %r(.+/Calls.*)
-      params = HTTP::Params.parse(request.body.not_nil!.gets_to_end)
-      @verboice_url = params["Url"]
+      account_sid = (/\/Accounts\/(.+)\/Calls.*/.match(request.path).try &.[1])
+      unless account_sid
+        plain_response(context, 400, "Account sid param is missing in the URL")
+        return
+      end
+      unless request.body
+        plain_response(context, 400, "Request body is missing")
+        return
+      end
+      body = request.body.not_nil!.gets_to_end
+      if body.blank?
+        plain_response(context, 400, "Request body is missing")
+        return
+      end
+      params = HTTP::Params.parse(body)
+      unless params.has_key?("Url")
+        plain_response(context, 400, "Body param 'Url' is missing")
+        return
+      end
+      verboice_url = params["Url"]
+      unless params.has_key?("From")
+        plain_response(context, 400, "Body param 'From' is missing")
+        return
+      end
+      from = params["From"]
+      unless params.has_key?("From")
+        plain_response(context, 400, "Body param 'To' is missing")
+        return
+      end
+      to = params["To"]
       context.response.status_code = 201
       context.response.content_type = "application/json"
       response = {"sid" => UUID.random().to_s()}
       response.to_json(context.response)
-    when %r(^/add$)
-      @nums << request.query_params["num"].to_i
-      context.response.status_code = 200
-      context.response.content_type = "text/plain"
-      context.response.puts "ok"
-    when %r(^/get$)
-      context.response.status_code = 200
-      @nums.to_json(context.response)
+      spawn do
+        sleep 1.seconds
+        request_body = "AccountSid=#{account_sid}&From=#{from}&To=#{to}&CallStatus=in-progress"
```

Comment:
Use `HTTP::Param` to build there strings https://crystal-lang.org/api/0.35.1/HTTP/Params.html